### DISCUSSION TITLE
Rsuite + Next.js + SASS

### DISCUSSION THREAD

**Discussion by nickcom** [rsuite/rsuite#1893] [Q&A] [answered]
I'm having issues getting Rsuite working with Next.js with SASS support.

Following the steps [here](https://rsuitejs.com/guide/use-next-app) I can get Rsuite and Next.js working. 

And, separately, following the steps [here](https://nextjs.org/docs/basic-features/built-in-css-support) I can get Next.js and SASS working.

But I can't work out how to get RSuite and Sass working together. Are there any tips?

Currently my next.config.js file is the same as the Rsuite support doc:

```
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

module.exports = {
  webpack(config) {
    config.module.rules.push({
      test: /\.(le|c)ss$/,
      use: [
        MiniCssExtractPlugin.loader,
        {
          loader: 'css-loader'
        },
        {
          loader: 'less-loader',
          options: {
            sourceMap: true,
            lessOptions: {
              javascriptEnabled: true
            }
          }
        }
      ]
    });

    config.plugins.push(
      new MiniCssExtractPlugin({
        filename: 'static/css/[name].css',
        chunkFilename: 'static/css/[contenthash].css'
      })
    );

    return config;
  }
};

```

and the error is:

```
./styles/globals.scss
Module parse failed: Unexpected token (2:5)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| html,
> body {
|   padding: 0;
|   margin: 0;
```

I assume I need something in the next.config.js to handle scss files? Any tips? Thanks.

**Comment by SevenOutman** [comment]
Does adding [`saasOptions` property](https://nextjs.org/docs/basic-features/built-in-css-support#customizing-sass-options) to your `next.config.js` help?

**Reply by nickcom** [reply]
if I'm doing it correctly, no it doesn't change anything.

So I've followed the steps from '[use in next.js](https://rsuitejs.com/guide/use-next-app)' and confirmed the project works. 

And then I follow the 2 steps in '[Sass supprt](https://nextjs.org/docs/basic-features/built-in-css-support#sass-support)' to install sass and further modify the next.config.js.

If I add the _sassOptions_ at the top of next.config.js then I get the same error:

next.config.js:
```
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const path = require('path')

module.exports = {

  sassOptions: {
    includePaths: [path.join(__dirname, 'styles')],
  },

  webpack(config) {
    config.module.rules.push({
      test: /\.(le|c)ss$/,
      use: [
        MiniCssExtractPlugin.loader,
        {
          loader: 'css-loader'
        },
        {
          loader: 'less-loader',
          options: {
            sourceMap: true,
            lessOptions: {
              javascriptEnabled: true
            }
          }
        }
      ]
    });

    config.plugins.push(
      new MiniCssExtractPlugin({
        filename: 'static/css/[name].css',
        chunkFilename: 'static/css/[contenthash].css'
      })
    );

    return config;
  }
};
```

error:
```
./styles/globals.scss
Module parse failed: Unexpected token (2:5)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| html,
> body {
|   padding: 0;
|   margin: 0;
```

./styles/globals.scss:
```
html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}
```

**Reply by SevenOutman** [reply]
Which version of RSuite and Next.js are you using?

**Comment by nickcom** [comment]
all just run right now, fresh install. 

RSuite: 4.10.2
Next.js: 11.1.2

**Comment by SevenOutman** [comment, marked as answer]
You should be getting warnings from Next.js going:

    Warning: Built-in CSS support is being disabled due to custom CSS configuration being detected.
    See here for more info: https://nextjs.org/docs/messages/built-in-css-disabled

A custom CSS configuration (in this case the rule for `.less` files) disables the built-in CSS support of Next.js, along with the built-in Sass support. So you have to setup Sass support yourself if you need to work with both Less and Sass.

**Reply by nickcom** [reply]
thanks, I suspected something similar. 

This ended up working for me. Maybe an admin could consider adding this to the next.js documentation page to help others?

1) install saas-loader: `npm install sass-loader sass webpack --save-dev`
2) update next.config.js like this:

```
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

module.exports = {

  webpack(config) {
    config.module.rules.push(
      {
      test: /\.(le|c)ss$/,
      use: [
        MiniCssExtractPlugin.loader,
        {
          loader: 'css-loader'
        },
        {
          loader: 'less-loader',
          options: {
            sourceMap: true,
            lessOptions: {
              javascriptEnabled: true
            }
          }
        }
      ]
    },
    {
      test: /\.(sa|sc)ss$/,
      use: [
        MiniCssExtractPlugin.loader,
        {
          loader: 'css-loader'
        },
        {
          loader: 'less-loader',
          options: {
            sourceMap: true,
            lessOptions: {
              javascriptEnabled: true
            }
          }
        },
        {
            loader: 'sass-loader',
            options: {
                sourceMap: true
            }
        }
      ]
    }
    );

    config.plugins.push(
      new MiniCssExtractPlugin({
        filename: 'static/css/[name].css',
        chunkFilename: 'static/css/[contenthash].css'
      })
    );

    return config;
  }
};

**Reply by SevenOutman** [reply]
You need `sass-loader` to handle Sass files in webpack, and you can find in its [documentation](https://www.npmjs.com/package/sass-loader) how to install and set it up.

BTW, you may simply write your styles in Less instead of Sass so that you don’t have to setup Sass support additionally.

If you actually don’t work with Less, you could [import compiled RSuite CSS](https://rsuitejs.com/guide/usage/) instead of Less so that you don’t have to bother tuning webpack.

```jsx 
import 'rsuite/dist/styles/rsuite-default.css';
```
